devwilliamy/coverland
# Submodel Search Params Not Refreshing Page & Adding IsComplete checks

Unfortunately two changes in this PR I should have made the other one a different PR

https://app.clickup.com/t/86889zcb1

## Change 1: Submodel Search Params Reload
When on an incomplete year page or on a submodel/submodel2 page, when you try to edit the vehicle and use the same type, make, model, and year parameters, the selected product won't update.

This is caused by Next.js thinking there are certain components that just use the search params and just does a soft reload. 

In an ideal world, this is the behavior we want and what should happen is that `selectedProduct` and `modelData` get updated depending on the submodel search params. I think the steps to do this is we have to keep a copy of the initial model data because we filter down on it based on submodel. If a search param is added in EditVehicleDropdown, then we just need to setSelectedProduct and setModelData based on it. If we change to another submodel, then we take the initialModelData and setModelData based on the selected submodel

However, for speed, I'm forcing a full reload of the page when submodel/submodel2 is added to ensure when a submodel is selected the proper selectedProduct is updated.

## How to Test

It is important that type, make, model, year are the SAME when making your selection. If any type, make, model, year is different, Next will detect a change in the URL and actually refresh the page, which is not the issue.

(year just needs to be in the same parent_generation range)

### Starting from year page -> submodel
- Go to an incomplete year page (find something with a submodel, remove the submodel from URL and just be on the year page)
- Use edit vehicle / car selector to fully select the car (ensure type, make, model, year are the same)
- Add to cart
- Verify the selected product is correct
- Attempt with not just the default color
- Repeat for car-cover/seat-cover whichever one you didn't do

### Changing between different submodels
- Go to a submodel page
- Use edit vehicle to change submodels (ensure type, make, model, year are the same)
- Add to cart
- Verify the selected product is correct
- Attempt with not just the default color
- Repeat for car-cover/seat-cover whichever one you didn't do


## Known Issue
Page looks like it reloads twice which I think was something we didn't want before, but it's better than nothing

## Change 2: Add to Cart / Preorder showing too early

The Add To Cart button would show the wrong text on the year page when selection is incomplete. 
I updated the code to check if the selection is complete

## How to Test
- Test with a selection w/o submodels
- Test with selection w/ submodel
- Test with selection w/ submodel2

Things to look out for:
- If not complete, shouldn't say preorder, and shipping details shouldn't say shipped when restocked
- Should say "Find your customer cover" or what not on incomplete pages
- Ensure you're able to add to cart on complete pages
